Fix auto fire again

kevingranade opened the pull request:
At some point the code handling "wild shots" that hit something other than the intended target became broken such that the chance of hitting a monster between the shooter and target was the same as the chance of hitting the target.  That is fixed by   e9bddff.
While I was at it I finally extended the trajectory of shots to the full range of the gun, so now you'll spray wild shots all over the place behind your target if you miss.  This is very beneficial when firing into a crowd of zombies, as you have a decent chance of hitting something.  To support this I added a range member to the projectile struct that gets passed between the various shooting methods.
As a quality of life fix I adjusted the code that draws bullet glyphs intermittently to widen the gaps as the bullets get further away from the shooter, so misses shouldn't take much longer than hits to animate.

DeadLeaves commented:
Hopefully this will make burst more viable when clearing crowds.

Coolthulhu commented on src/ranged.cpp:
```diff
@@ -150,8 +150,11 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
     tripoint prev_point = source;
+    std::vector<tripoint> trajectory_extension = continue_line( trajectory, proj_arg.range );
+    trajectory.resize( trajectory.size() + trajectory_extension.size() );
+    trajectory.insert( trajectory.end(), trajectory_extension.begin(), trajectory_extension.end() );
```
Is this correct? The way I understand it, it will insert `trajectory_extension.size()` default value tripoints between the end of original trajectory and the new extended one.
Shouldn't it be `reserve` instead of `resize`?

kevingranade replied:
Indeed, right now it's going to process the square past the target trajectory_extension.size() times.

kevingranade commented:
Fixed my mistake in calling the wrong vector method, then I noticed that continue line was adding the range of the gun to the distance to the target, that won't do so I fixed that.  Then I noticed it wasn't respecting trigdist, so fixed that.  Finally noticed if you fire wildly at a target at the edge of your range, the "wild shot" code would pick a target point outside of your firing range, so fixed that.  while I was at it I rebased on most recent master.
